sagemaker/launch/realtime_entry_points/node_prediction_entry.py

```diff
@@ -0,0 +1,386 @@
+                res = res_msg.mismatch_target_nid(target_nid)
+                return res
+            else:
+                graph_target_nid = raw_node_id_maps[target_ntype][target_nid]
+                graph_target_nids.append(graph_target_nid)
+        else:
+            orig_target_nids = []
+            orig_target_nids.append(target_nid)
+            graph_target_nids = []
+            # target id is not in the subgraph
+            if raw_node_id_maps[target_ntype].get(target_nid, None) is None:
+                res = res_msg.mismatch_target_nid(target_nid)
+                return res
+            else:
+                graph_target_nid = raw_node_id_maps[target_ntype][target_nid]
+                graph_target_nids.append(graph_target_nid)
+                target_mapping_dict[target_ntype] = (orig_target_nids, graph_target_nids)
+
+    data = {DATA: dgl_graph, TARGETS: target_mapping_dict}
+    res = res_msg.success(data=data)
+
+    e_t = dt.now()
+    diff_t = int((e_t - s_t).microseconds / 1000)
+    logging.info(f'--input_fn: used {diff_t} ms ...')
+
+    return res
+
+def predict_fn(input_data, model):
+    """ Make prediction on the given subgraph for the given targets with the loaded model.
+    
+    Parameters
+    ----------
+    input_data: dict
+        The input data is a GSRealTimeInferenceResponseMessage, where 'status_code': 200,
+        'message': 'something', 'data': {'data': dgl_graph, 'targets': target_dict}}.
+        The target_dict is a dictionary whose keys are target node types, and keys are tuples
+        where the 1st element is a list of the original node IDs, and the 2nd element is a list
+        of the new DGL graph integer IDs. The input_data comes from the input_fn functioin.
+    model: GraphStorm model
+        A GraphStorm model loaded from the model_fn() function.
+    """
+    logging.info('-- START prediction... ')
+    s_t = dt.now()
+
+    res = {}
+
+    # Handle payload errors from the input_fn
+    if input_data.status_code != 200:
+        return input_data.to_dict()
+
+    # extract the data
+    dgl_graph = input_data.data[DATA]
+    target_dict = input_data.data[TARGETS]
+
+    # extract the DGL graph nids from target ID mapping dict
+    target_nids = {}
+    for ntype, (_, dgl_nids) in target_dict.items():
+        target_nids[ntype] = dgl_nids
+
+    # Use load model configuration to intialize an inferrer
+    global GS_CONFIG
+    gs_config = GS_CONFIG
+
+    try:
+        # setup device
+        setup_device(0)
+        device = get_device()
+
+        # initialize a GS real-time inferrer
+        inferrer = GSGnnNodePredictionRealtimeInferrer(model)
+        inferrer.setup_device(device)
+        # initialize a GS real-time dataLoader
+        dataloader = GSgnnRealtimeInferNodeDataLoader(dgl_graph,
+                                                      target_nids,
+                                                      gs_config.num_layers)
```

Comment:
Shuffle = False, right?

Reply:
Yes, the dataloader force to use Shuffle=False internally.